internal/yaml/yaml_test.go

```diff
@@ -1,13 +1,185 @@
 package yaml
 
 import (
+	"bytes"
+	"os"
+	"path/filepath"
 	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/require"
 	"gopkg.in/yaml.v3"
 )
 
+func TestRead(t *testing.T) {
+	abDir, err := os.MkdirTemp("", "ab")
```

Comment:
Nit: with the number of test cases, it may be appealing to put them in individual descriptive `t.Run` cases to more easily spot what breaks.

Reply:
I typically do for table-driven tests. I didn't in this particular case since the setup of each case builds on the setup of the previous case.

I realized only after you said something that I could leave setup _outside_ of `t.Run` and  wrap `t.Run` only around groups of assertions and that wouldn't prevent iteratively building the more complex cases.

All set and thank you for a good idea.

Reply:
Only spotted this now 🤦 

But if you use `t.TempDir()`, you can get away with not doing any `defer` calls to clean things up _and_ the clean up will still succeed if the test case fails, instead of it accumulating directories in temp.

Reply:
Forgot about that!

Also found another edge case involving symlinks... switching this to draft for now.

Reply:
Ok... switched to `t.TempDir()`. Thanks again for the reminder.

Two edge cases I found:

1. If we end up reading a symlink that points to a directory _containing_ what we're reading, the existing logic for preventing infinite recursion will work, but duplicates may be read in the process.

2. If we deliberately did `yaml.Read("foo", "foo")`, we'll get all the resources twice.

Both problems can be solved by keeping track of paths we've read and short-circuiting reads of things already read.

Reply:
These two new edge cases are addressed in 26d3e66cbae702a8cbb58748b357e6d2786dd022